src/react-components/src/components/02-molecules/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,82 @@
+// MF-46 Search Bar with Search By Parameter
+
+import * as React from "react";
+import bem from "../../../utils/bem";
+import FormDropdown from "../../01-atoms/Forms/FormDropdown";
+import TextField from "../../01-atoms/Forms/TextField";
+import { Button } from "../../..";
+
+export interface SearchBarProps {
+  searchBarId: string;
+  dropdownId?: string;
+  dropdownOptions?: string[];
+  selectedField?: string;
+  searchValue?: string;
+  placeHolder?: string;
+  searchButtonAttributes?: {};
+  placeholderText?: string;
+  textFieldAttributes?: {};
+  hasError?: boolean;
+  errorMessage?: string;
+  selectBlurHandler?: (event: React.MouseEvent) => void;
+  selectChangeHandler?: (event: React.MouseEvent) => void;
+  searchSubmitHandler: (event: React.MouseEvent) => void;
+  searchChangeHandler?: (event: React.FormEvent) => void;
+}
+
+export default function SearchBar(props: SearchBarProps) {
+
+  const { searchBarId, dropdownId, dropdownOptions, selectedField, placeholderText, hasError, errorMessage,
+    selectBlurHandler, searchSubmitHandler, selectChangeHandler, searchChangeHandler } = props;
+
+  if (dropdownOptions) {
+    if (!(dropdownId && selectBlurHandler && selectChangeHandler)) {
+      throw new Error("If dropdownOptions are passed, dropdownId, selectChangeHandler, and selectBlurHandler must also be passed");
+    }
+  }
+  let modifiers = [];
+  if (hasError) {
+    if (!errorMessage) {
+      throw new Error("If there is an error, there must also be an error message");
+    }
+    modifiers.push("error");
+  }
+
+  let searchbar__base_class = "search-bar";
+
+  let textfieldProps = {
+    labelId: searchBarId,
+    onChange: searchChangeHandler,
+    isRequired: true,
+    blockName: searchbar__base_class,
+    placeholderText: placeholderText,
+    modifiers: modifiers,
+  };
+
+  let buttonProps = {
+    id: searchBarId,
+    callback: searchSubmitHandler,
+    blockName: searchbar__base_class,
+    content: "Search",
+    type: "filled",
+    iconPosition: "left",
+    iconName: "search-small",
+    iconDecorative: true,
+  };
+
+  return <form className={bem(searchbar__base_class, modifiers)} id="searchBarId" aria-label="Search for keyword, author, title, or subject">
```

Comment:
This trips me up but having a static `id` can be troublesome if the component gets rendered more than once, the accessibility error is having duplicate ids on the page. For the `SearchBar` I imagine there will only be one, a main one, but if the design of any page reuses this, then the `id` should be dynamic.